docs/cloudsync_mqtt.md

```diff
@@ -0,0 +1,272 @@
+
+commonName           = Common Name (e.g. server FQDN or YOUR name)
+commonName_default   = Test Server
+
+emailAddress         = Email Address
+emailAddress_default = test@example.com
+
+####################################################################
+[ server_req_extensions ]
+
+subjectKeyIdentifier = hash
+basicConstraints     = CA:FALSE
+keyUsage             = digitalSignature, keyEncipherment
+subjectAltName       = @alternate_names
+nsComment            = "OpenSSL Generated Certificate"
+
+####################################################################
+[ alternate_names ]
+
+DNS.1  = example.com
+DNS.2  = www.example.com
+DNS.3  = mail.example.com
+DNS.4  = ftp.example.com
+
+```
+Then, create the server certificate request
+```
+$ openssl req -config openssl-server.cnf -newkey rsa:2048 -sha256 -nodes -out servercert.csr -outform PEM
+```
+After this command executes, you will have a request in servercert.csr and a private key in serverkey.pem
+
+Next, you have to sign it with your CA.open openssl-ca.cnf and add the following two sections.
+```
+####################################################################
+[ signing_policy ]
+countryName            = optional
+stateOrProvinceName    = optional
+localityName           = optional
+organizationName       = optional
+organizationalUnitName = optional
+commonName             = supplied
+emailAddress           = optional
+
+####################################################################
+[ signing_req ]
+subjectKeyIdentifier   = hash
+authorityKeyIdentifier = keyid,issuer
+basicConstraints       = CA:FALSE
+keyUsage               = digitalSignature, keyEncipherment
+```
+Add the following to the [ CA_default ] section of openssl-ca.cnf.
+```
+base_dir      = .
+certificate   = $base_dir/cacert.pem   # The CA certifcate
+private_key   = $base_dir/cakey.pem    # The CA private key
+new_certs_dir = $base_dir              # Location for new certs after signing
+database      = $base_dir/index.txt    # Database index file
+serial        = $base_dir/serial.txt   # The current serial number
+
+unique_subject = no  # Set to 'no' to allow creation of
+                     # several certificates with same subject.
+
+```
+Third, touch index.txt and serial.txt:
+```
+$ touch index.txt
+$ echo '01' > serial.txt
+```
+Then, perform the following:
+```
+$ openssl ca -config openssl-ca.cnf -policy signing_policy -extensions signing_req -out servercert.pem -infiles servercert.csr
+```
+
+### Copying the Certificates
+1. Copy the servercert.pem ,serverkey.pem and cacert.pem file to /etc/mosquitto/certs
+2. Mention the path of the certs in the configuration file of mosquitto
+3. Copy the cacert.pem file in /var/edge-orchestration/mqtt/certs folder in client side
+4. In the broker For each user paswword can be generated. If the password is generated then oassword file path can be specified in the configuration file with allow_anonymous made to false.
```

Comment:
What does this mean? `In the broker For each user password can be generated`

```suggestion
4. In the broker For each user password can be generated. If the password is generated then password file path can be specified in the configuration file with allow_anonymous made to false.
```

Reply:
Re framed the statements.